auto_gidn.c

```diff
@@ -0,0 +1,11 @@
+#include "uidgid.h"
+#include "auto_uids.h"
+#include "auto_users.h"
+
+int auto_gidn;
+
+__attribute__((constructor)) static void
```

Comment:
Reading this, I like how it self-describes, how it works, and how it helps simplify the other changes in this PR.

I worry about compatibility. Besides gcc and clang, which compilers are known to support this? Besides ELF and Mach-O, which executable formats are known to support this? Is this defined in a particular C language standard that we're now requiring?

Reply:
We defined this line of inquiry out of existence by removing the constructor attribute from the code.